python/tank/util/local_file_storage.py

```diff
@@ -164,6 +164,10 @@ def get_site_root(cls, hostname, path_type, generation=CORE_V18):
                            which is the current generation of paths.
         :return: Path as string
         """
+        if hostname is None:
```

Comment:
I always feels it's cleaner if errors that we detect are raised as TankErrors since it allows a dev to know without even looking at Toolkit's code that the bug is something we detected and not something they've hit because there is a bug in our code. Us raising ValueError is indistinguishable from us misusing a value.